cmd/podman/varlink/io.podman.varlink

```diff
@@ -224,117 +224,137 @@ type Sockets(
-    conmon_pidfile: string,
-    cgroup_parent: string,
-    command: []string,
-    detach: bool,
-    devices: []string,
-    dns_opt: []string,
-    dns_search: []string,
-    dns_servers: []string,
-    entrypoint: []string,
-    env:  [string]string,
-    exposed_ports: []string,
-    gidmap: []string,
-    group_add: []string,
-    host_add: []string,
-    hostname: string,
-    image: string,
-    image_id: string,
-    init: bool,
-    init_path: string,
-    builtin_imgvolumes: []string,
-    id_mappings: IDMappingOptions,
-    image_volume_type: string,
-    interactive: bool,
-    ipc_mode: string,
-    labels: [string]string,
-    log_driver: string,
-    log_driver_opt: []string,
+type CreateResult (
+    changed: bool,
     name: string,
-    net_mode: string,
-    network: string,
-    pid_mode: string,
-    pod: string,
-    privileged: bool,
-    publish: []string,
-    publish_all: bool,
-    quiet: bool,
-    readonly_rootfs: bool,
-    resources: CreateResourceConfig,
-    rm: bool,
-    shm_dir: string,
-    stop_signal: int,
-    stop_timeout: int,
-    subuidmap: string,
-    subgidmap: string,
-    subuidname: string,
-    subgidname: string,
-    sys_ctl: [string]string,
-    tmpfs: []string,
-    tty: bool,
-    uidmap: []string,
-    userns_mode: string,
-    user: string,
-    uts_mode: string,
-    volumes: []string,
-    work_dir: string,
-    mount_label: string,
-    process_label: string,
-    no_new_privs: bool,
-    apparmor_profile: string,
-    seccomp_profile_path: string,
-    security_opts: []string
+    valueType: string
 )
 
-# CreateResourceConfig is an input structure used to describe host attributes during
-# container creation.  It is only valid inside a [Create](#Create) type.
-type CreateResourceConfig (
-    blkio_weight: int,
-    blkio_weight_device: []string,
-    cpu_period: int,
-    cpu_quota: int,
-    cpu_rt_period: int,
-    cpu_rt_runtime: int,
-    cpu_shares: int,
-    cpus: float,
-    cpuset_cpus: string,
-    cpuset_mems: string,
-    device_read_bps: []string,
-    device_read_iops: []string,
-    device_write_bps: []string,
-    device_write_iops: []string,
-    disable_oomkiller: bool,
-    kernel_memory: int,
-    memory: int,
-    memory_reservation: int,
-    memory_swap: int,
-    memory_swappiness: int,
-    oom_score_adj: int,
-    pids_limit: int,
-    shm_size: int,
-    ulimit: []string
+type CreateString (
+    cr: CreateResult,
+    value: string
 )
 
-# IDMappingOptions is an input structure used to described ids during container creation.
-type IDMappingOptions (
-    host_uid_mapping: bool,
-    host_gid_mapping: bool,
-    uid_map: IDMap,
-    gid_map: IDMap
+type CreateStringSlice (
+    cr: CreateResult,
+    value: []string
 )
 
-# IDMap is used to describe user name spaces during container creation
-type IDMap (
-    container_id: int,
-    host_id: int,
-    size: int
+type CreateInt(
+    cr: CreateResult,
+    value: int
+)
+
+type CreateBool(
```

Comment:
In a longer term, our next step should be to remove the CreateResult part. This is going to be a major effort - we need to split the logic into frontend and backend, to remove any need for IsSet checks in the backend - but I'm confident it can be done.

Reply:
agree, and perhaps the merging of this PR will inspire me to do it sooner than later.